src/Chapter2.hs

```diff
@@ -825,42 +664,41 @@ Infinity Stone!
 
 {- |
 =⚔️= Task 11
-
 Rotating a list by a single element is the process of moving the first
 element of the list to the end.
-
 Implement a function to rotate a given finite list by N elements. Try
 to do it more efficiently than rotating by a single element N times.
-
 On invalid input (negative rotation coefficient) it should return an empty
 list.
-
 >>> rotate 1 [1,2,3,4]
 [2,3,4,1]
 >>> rotate 3 [1,2,3,4]
 [4,1,2,3]
-
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+rotate :: a -> [a] -> [a]
+rotate n l
+  | n < 0 || null l = []
+  | otherwise = drop n (take (length 1+n) (cycle l))
```

Comment:
That's a correct solution! Though, it does a little bit more than work when the given number is big and the list is small. You will cycle through the list a lot. Maybe you can think of a way to optimize this :slightly_smiling_face: